src/app/space/app-launcher/services/app-launcher-gitprovider.service.ts

```diff
@@ -85,18 +85,21 @@ export class AppLauncherGitproviderService implements GitProviderService {
   getGitHubDetails(): Observable<GitHubDetails> {
     return this.getGitHubUserData().pipe(mergeMap(user => {
       if (user && user.login) {
-        let orgs = [];
+        this.gitHubUserLogin = user.login;
+        let orgs: { [name: string]: string } = {};
```

Comment:
Is this `Array` of strings as a key for `object`? Can we add a tyep interface for this? And why do we need such a complex data structure to represent `orgs`?

Reply:
it's a "Index Signature" it means the key has to be a string and the value as well: https://basarat.gitbooks.io/typescript/docs/types/index-signatures.html

Reply:
Ohh nice. Didn't knew you could define type of index in an object as well. Thanks for sharing.